src/lib/chips/chip.ts

```diff
@@ -123,15 +164,76 @@ export class MdChip implements Focusable, OnInit, OnDestroy {
     this.onFocus.emit({chip: this});
   }
 
+  /**
+   * Allows for programmatic removal of the chip. Called by the MdChipList when the DELETE or
+   * BACKSPACE keys are pressed.
+   *
+   * Note: This only informs any listeners of the removal request, it does **not** actually remove
+   * the chip from the DOM.
+   */
+  remove(): void {
+    if (this.removable) {
+      this.onRemove.emit({chip: this});
+    }
+  }
+
   /** Ensures events fire properly upon click. */
   _handleClick(event: Event) {
     // Check disabled
+    if (this._checkDisabled(event)) {
+      return;
+    }
+
+    event.preventDefault();
+    event.stopPropagation();
+
+    this.focus();
+  }
+
+  /** Handle custom key presses. */
+  _handleKeydown(event: KeyboardEvent) {
+    if (this._checkDisabled(event)) {
+      return;
+    }
+
+    switch (event.keyCode) {
+      case DELETE:
+      case BACKSPACE:
+        // If we are removable, remove the focused chip
+        if (this.removable) {
+          this.onRemove.emit();
+        }
+
+        // Always prevent so page navigation does not occur
+        event.preventDefault();
+        break;
+      case SPACE:
+        // If we are selectable, toggle the focused chip
+        if (this.selectable) {
+          this.toggleSelected();
+        }
+
+        // Always prevent space from scrolling the page since the list has focus
+        event.preventDefault();
+        break;
+    }
+  }
+
+  /**
+   * Sets whether or not this chip is displaying a remove icon. Adds/removes the
+   * `md-chip-has-remove-icon` class.
+   */
+  _setHasRemoveIcon(value: boolean) {
+    this._hasRemoveIcon = value;
+  }
+
+  protected _checkDisabled(event: Event): boolean {
```

Comment:
Can we omit this method? It's name doesn't really tell you what it does and it's just a big side-effect